### PR TITLE
Enable CAE by Default

### PR DESCRIPTION
**Pull request by shemogumbe**
Fixes https://github.com/microsoft/kiota-authentication-azure-python/issues/407

### CODE REVIEW

File: kiota_authentication_azure/azure_identity_authentication_provider.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def __init__(
         self,
         credentials: Union["TokenCredential", "AsyncTokenCredential"],
         options: Optional[Dict] = None,
+        is_cae_enabled: bool = True,
```
Comment (baywet) [inline kiota_authentication_azure/azure_identity_authentication_provider.py:18]: inserting a parameter in the middle like that will break people who do not rely on named parameters and simply rely on the order of things.
I believe this new parameter should be at the end.

File: kiota_authentication_azure/azure_identity_access_token_provider.py [hunk 2]
```diff
@@ -45,10 +47,14 @@ def __init__(
...
         self._options = options
+        self._is_cae_enabled = is_cae_enabled
         self._allowed_hosts_validator = AllowedHostsValidator(allowed_hosts)
 
     async def get_authorization_token(
-        self, uri: str, additional_authentication_context: Dict[str, Any] = {}
+        self,
+        uri: str,
+        additional_authentication_context: Dict[str, Any] = {},
+        is_cae_enabled: bool = True
```
Comment (baywet) [inline kiota_authentication_azure/azure_identity_access_token_provider.py:57]: adding a parameter to this method deviates from the interface contract. We should remove it and rely on the property/field only instead.

**Review by baywet — CHANGES_REQUESTED** [review]

File: kiota_authentication_azure/azure_identity_access_token_provider.py (hunk 2 above)
Comment (shemogumbe) [reply kiota_authentication_azure/azure_identity_access_token_provider.py]: I have removed it altogether, and relied only on the value from the constructor

File: kiota_authentication_azure/azure_identity_access_token_provider.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ def __init__(
         options: Optional[Dict],
         scopes: List[str] = [],
         allowed_hosts: List[str] = [],
+        is_cae_enabled: Optional[bool] = True,
```
Comment (baywet) [inline kiota_authentication_azure/azure_identity_access_token_provider.py:36]: doesn't the default value make it optional for the caller already?

File: kiota_authentication_azure/azure_identity_access_token_provider.py (hunk 2 above)
Comment (baywet) [reply kiota_authentication_azure/azure_identity_access_token_provider.py]: thanks, resolving

File: kiota_authentication_azure/azure_identity_access_token_provider.py (hunk 3 above)
Comment (shemogumbe) [reply kiota_authentication_azure/azure_identity_access_token_provider.py:36]: Yes, I think we should not make it Optionally and use default value of true to enable it by default without user having to explicitly set

**Review by baywet — DISMISSED** [review]
Thank you for making the changes!
A version bump and we should be good to go.

**Pull request merged**